src/priceClient/adapters/defiLlama.ts

```diff
@@ -54,19 +54,23 @@ export class PriceFeed extends BaseHTTPAdapter implements PriceFeedAdapter {
     if (currency != "usd") throw new Error(`Currency ${currency} not supported by ${this.name}`);
 
     const path = "prices/current/" + addresses.map((address) => `ethereum:${address}`).join();
-    const tokenPrices: unknown = await this.query(path, {});
-    if (!this.validateResponse(tokenPrices))
-      throw new Error(`Unexpected ${this.name} response: ${JSON.stringify(tokenPrices)}`);
-
-    return addresses.map((address: string) => {
-      const tokenPrice = tokenPrices.coins[`ethereum:${address}`];
-      if (tokenPrice === undefined) throw new Error(`Token ${address} missing from ${this.name} response`);
-
-      if (tokenPrice.confidence < this.minConfidence)
-        throw new Error(`Token ${address} has low confidence score (${tokenPrice.confidence})`);
-
-      return { address, price: tokenPrice.price, timestamp: tokenPrice.timestamp };
-    });
+    const response: unknown = await this.query(path, {});
+    if (!this.validateResponse(response))
+      throw new Error(`Unexpected ${this.name} response: ${JSON.stringify(response)}`);
+
+    // Normalise the address format: "etherum:<address>" => "<address>".
+    const tokenPrices: { [address: string]: DefiLlamaTokenPrice } = Object.fromEntries(
+      Object.entries(response.coins).map(([identifier, tokenPrice]) => [identifier.split(":")[1], tokenPrice])
+    );
+
+    return Object.entries(tokenPrices)
+      .filter(([address, tokenPrice]) => {
+        return addresses.includes(address) && tokenPrice.confidence >= this.minConfidence;
```

Comment:
Are we sure the casing is going to match between addresses and the response? I would suggest ensuring that both are formatted the same before comparing, since both are coming from outside sources.

Reply:
Fair point. DefiLlama seem to respect the case per current implementation, but CoinGecko does not (i.e. it seems to always return a lower-case address). I've modified the adapter implementations in https://github.com/across-protocol/sdk-v2/pull/128/commits/62fe2cef9398ef1543701985064b6b7a8fc49301 in order to force lower-case comparisons.